hypervisor/include/arch/x86/timer.h

```diff
@@ -48,19 +72,57 @@ static inline void initialize_timer(struct hv_timer *timer,
 	}
 }
 
+/**
+ * @brief Check a timer whether expired.
+ *
+ * @param[in] timer Pointer to timer.
+ *
+ * @return true if the timer is expired
+ * @return false if the timer is not expired.
+ */
 static inline bool timer_expired(const struct hv_timer *timer)
 {
 	return ((timer->fire_tsc == 0UL) || (rdtsc() >= timer->fire_tsc));
 }
 
-/*
- * Don't call add_timer/del_timer in the timer callback function.
+/**
+ * @brief Add a timer.
+ *
+ * @param[in] timer Pointer to timer.
+ *
+ * @return 0       - on success.
+ * @return -EINVAL - \p timer has an invalid value.
+ *
+ * @remark Don't call it in the timer callback function or interrupt content.
  */
 int add_timer(struct hv_timer *timer);
+
+/**
+ * @brief Delet a timer.
+ *
+ * @param[in] timer Pointer to timer.
+ *
+ * @remark Don't call it in the timer callback function or interrupt content.
+ */
 void del_timer(struct hv_timer *timer);
 
+/**
+ * @brief Initialize timer.
+ */
```

Comment:
@junjiemao1 , do we want to have both a `@param None` and `@return None` directives here?

Same question for the other functions below.

Reply:
Makes sense, thanks @junjiemao1